File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    result = [];
    temp = [];

    for (i = 0; i < score.length; i++) {
        if (i < k) {
            temp.push(score[i]);
            temp.sort((a, b) => a - b);
            result.push(temp[0]);
        } else if (i >= k && score[i] > temp[0]) {
            temp.shift();
            temp.push(score[i]);
            temp.sort((a, b) => a - b);
            result.push(temp[0]);
        } else if (i >= k && score[i] <= temp[0]) {
            result.push(temp[0]);
        }
    }
    return result;
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    result = []
    temp = []
    
    for i in range(0, len(score)):
        if(i<k):
            temp.append(score[i])
            temp = sorted(temp)
            result.append(temp[0])
        
        if(i >= k and score[i] > temp[0]):
            temp.pop(0)
            temp.append(score[i])
            temp = sorted(temp)
            result.append(temp[0])
        elif(i >= k and score[i] <= temp[0]):
            result.append(temp[0])
            
    return result